test/dynamo/test_misc.py

```diff
@@ -11534,6 +11534,19 @@ def get_rng():
         self.assertEqual(expected, actual)
         self.assertGreater(po.call_count, 0)
 
+    def test_data_ptr_graph_break(self):
+        def f(a, b):
+            # builtin + not implemented for DataPtrVariable
+            return a.data_ptr() + b.data_ptr()
+
+        a = torch.randn(4)
+        b = torch.randn(5)
+
+        expected = f(a, b)
+        actual = torch.compile(f, backend="eager")(a, b)
+
```

Comment:
Can we do an assertion that there was a graph break?

Reply:
I've added another test (`test_data_ptr_graph_break_aten`) in which I can assert on the graph break. In the original test (now named `test_data_ptr_graph_break_builtin`), `counters["graph_break"]` remains empty, although I can see a graph break in the `TORCH_LOGS="+dynamo"`. Would this state be sufficient for testing?

Reply:
Yes, with `fullgraph=True`, we get this error:

```
torch._dynamo.exc.Unsupported: builtin: add [<class 'torch._dynamo.variables.tensor.DataPtrVariable'>, <class 'torch._dynamo.variables.tensor.DataPtrVariable'>] False
```

As @williamwen42 has pointed out:

> it looks like only certain classes of graph breaks actually contribute to counters["graph_break"] - graph breaks caused by STORE_ATTRs and function call bytecode. in your graph break case, BINARY_ADD causes the graph break, which is not one of those classes. it seems generally that graph breaks that cause the entire frame to be skipped don't count.

I've added `self.assertRaises` with `fullgraph=True` to the test to make sure there is a graph break. Please let me know if this is good enough.

Reply:
Weird, thanks for looking into this (and adding that test)